src/GitReleaseManager.Core/Provider/GitLabProvider.cs

```diff
@@ -0,0 +1,435 @@
+            });
+        }
+
+        public Task PublishReleaseAsync(string owner, string repository, string tagName, Release release)
+        {
+            return ExecuteAsync(async () =>
+            {
+                var releaseClient = _gitLabClient.GetReleases(GetGitLabProjectId(owner, repository));
+
+                var update = new ReleaseUpdate
+                {
+                    ReleasedAt = DateTime.UtcNow,
+                    TagName = tagName,
+                };
+
+                releaseClient.Update(update);
+            });
+        }
+
+        public Task UpdateReleaseAsync(string owner, string repository, Release release)
+        {
+            return ExecuteAsync(async () =>
+            {
+                var releaseClient = _gitLabClient.GetReleases(GetGitLabProjectId(owner, repository));
+
+                var update = new ReleaseUpdate
+                {
+
+                    Description = release.Body,
+                    ReleasedAt = release.Draft ? DateTime.UtcNow.AddYears(1) : DateTime.UtcNow,
+                    Name = release.Name,
+                    TagName = release.TagName,
+                    Milestones = new string[] { release.TagName },
+                };
+
+                releaseClient.Update(update);
+            });
+        }
+
+        public RateLimit GetRateLimit()
+        {
+            // TODO: There doesn't currently seem to be a way to get the remaining
+            // rate limit for GitLab using the library we are using, so for now,
+            // let's just hard code it.
+            return new RateLimit { Limit = 600, Remaining = 100 };
+        }
+
+        public string GetMilestoneQueryString()
+        {
+            return "sort=due_date_desc&state=closed";
+        }
+
+        public string GetIssueType(Issue issue)
+        {
+            return issue.IsPullRequest ? "Merge Request" : "Issue";
+        }
+
+        private int GetGitLabProjectId(string owner, string repository)
+        {
+            if (_projectId.HasValue)
+            {
+                return _projectId.Value;
+            }
+
+            var projectName = string.Format(CultureInfo.InvariantCulture, "{0}/{1}", owner, repository);
+            var project = _gitLabClient.Projects[projectName];
+            _projectId = project.Id;
+
+            return _projectId.Value;
+        }
+
+        private async Task ExecuteAsync(Func<Task> action)
+        {
+            try
+            {
+                await action().ConfigureAwait(false);
+            }
+            ////catch (Octokit.ForbiddenException ex)
+            ////{
+            ////    throw new ForbiddenException(ex.Message, ex);
+            ////}
+            ////catch (Octokit.NotFoundException ex)
+            ////{
+            ////    throw new NotFoundException(ex.Message, ex);
+            ////}
+            catch (Exception ex) when (!(ex is NotFoundException))
```

Comment:
We should probably also catch an [AggregateException](https://learn.microsoft.com/en-us/dotnet/api/system.aggregateexception?view=netstandard-2.0) which is commonly thrown in async functions.